views/js/runner/navigator/offlineNavigator.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Péter Halász <peter@taotesting.com>
+ */
+define([
+    'lodash',
+    'core/Promise',
+    'util/capitalize',
+    'taoQtiTest/runner/helpers/offlineJumpTable',
+    'taoQtiTest/runner/helpers/testContextBuilder'
+], function(
+    _,
+    Promise,
+    capitalize,
+    OfflineJumpTableHelper,
+    TestContextBuilder
+) {
+    'use strict';
+
+    return function offlineNavigatorFactory(itemStore) {
+        var testData,
+            testContext,
+            testMap,
+            offlineJumpTableHelper = new OfflineJumpTableHelper(itemStore);
```

Comment:
Caution you are using a factory as constructor.

Isn't the `OfflineJumpTableHelper` a helper? if so should not it be a static object instead? Having to instanciate a helper could lead to issues.